File: westcoast/src/main/java/frc/subsystems/drive/motorfactory/MotorFactoryHybrid.java
package frc.subsystems.drive.motorfactory;

import com.ctre.phoenix.motorcontrol.NeutralMode;
import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
import com.ctre.phoenix.motorcontrol.can.WPI_VictorSPX;
import edu.wpi.first.wpilibj.SpeedController;

public class MotorFactoryHybrid implements MotorFactory {
    public SpeedController create(int channelMain, int channel1, int channel2) {
        WPI_TalonSRX motorMain = new WPI_TalonSRX(channelMain);
        WPI_VictorSPX motor1 = new WPI_VictorSPX(channel1);
        WPI_VictorSPX motor2 = new WPI_VictorSPX(channel2);

        motor1.follow(motorMain);
        motor2.follow(motorMain);

        motorMain.configContinuousCurrentLimit(18);
        motorMain.configPeakCurrentLimit(20);
        motorMain.configPeakCurrentDuration(45);
        motorMain.enableCurrentLimit(true);
        motorMain.setNeutralMode(NeutralMode.Brake);

        return motorMain;
    }
}

File: westcoast/src/main/kotlin/frc/subsystems/drive/motorfactory/MotorFactoryHybrid.kt
package frc.subsystems.drive.motorfactory

import com.ctre.phoenix.motorcontrol.NeutralMode
import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX
import com.ctre.phoenix.motorcontrol.can.WPI_VictorSPX
import edu.wpi.first.wpilibj.SpeedController

class MotorFactoryHybrid : MotorFactory {
  override fun create(channelMain: Int, channel1: Int, channel2: Int): SpeedController {
    val motorMain = WPI_TalonSRX(channelMain)
    val motor1 = WPI_VictorSPX(channel1)
    val motor2 = WPI_VictorSPX(channel2)
    motor1.follow(motorMain)
    motor2.follow(motorMain)
    motorMain.configContinuousCurrentLimit(18)
    motorMain.configPeakCurrentLimit(20)
    motorMain.configPeakCurrentDuration(45)
    motorMain.enableCurrentLimit(true)
    motorMain.setNeutralMode(NeutralMode.Brake)
    return motorMain
  }
}